samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -132,7 +132,12 @@ def __init__(
         self._debug_args = debug_args
         self._debugger_path = debugger_path
         self._container_env_vars_file = container_env_vars_file
+
         self._parameter_overrides = parameter_overrides or {}
+        # Override certain CloudFormation pseudo-parameters based on values provided by customer
+        if self._aws_region:
```

Comment:
Why are we moving this out of the property? You don't set `self._aws_region` until line 143, so this is actually always `None`.

Reply:
Just noticed that too, fixed

Reply:
From what my understanding of the code, this is a part of `self._parameter_overrides` initialization process so it should happen in the constructor instead of in the property.